internal/util/util_test.go

```diff
@@ -157,3 +159,135 @@ func (t *UtilTest) TestResolveConfigFilePaths() {
 	AssertEq(nil, err)
 	ExpectEq(filepath.Join(homeDir, "test.txt"), mountConfig.LogConfig.FilePath)
 }
+
+func (t *UtilTest) TestRoundDurationToNextMultiple() {
+	cases := []struct {
+		input  time.Duration
+		factor time.Duration
+		output time.Duration
+	}{
+		{
+			input:  -time.Second - time.Nanosecond,
+			factor: time.Second,
+			output: -time.Second,
+		},
+		{
+			input:  -time.Second,
+			factor: time.Second,
+			output: -time.Second,
+		},
+		{
+			input:  -time.Second + time.Nanosecond,
+			factor: time.Second,
+			output: 0,
+		},
+		{
```

Comment:
can we add more +ve cases ? also, i believe the function will work for fractional factor also (e.g. dur = 1.3 seconds , factor = 1.5 seconds will give 1.5 seconds )? if yes, let's add 1-2 test cases for the same so that developer knows the capabilities of the function clearly.

Reply:
Added more tests, particularly for factor being a fractional multiple of a second. 
More tests can be added to test for scenarios around multiples of 1ns, 1us, 1ms, 1m, 1h etc. too. But that'll cause dozens of test. For now, I think factors of the order of a second generalizes to all such cases.

Reply:
Also added more details to the function header of RoundDurationToNextMultiple.